app/src/main/java/com/keylesspalace/tusky/ComposeActivity.java

```diff
@@ -1063,7 +1063,9 @@ private void addMediaToQueue(@Nullable String id, QueuedMedia.Type type, Bitmap
 
         if (item.readyStage != QueuedMedia.ReadyStage.UPLOADED) {
             waitForMediaLatch.countUp();
-            if (mediaSize > STATUS_MEDIA_SIZE_LIMIT && type == QueuedMedia.Type.IMAGE) {
+
+            if (type == QueuedMedia.Type.IMAGE &&
+                    (mediaSize > STATUS_MEDIA_SIZE_LIMIT || MediaUtils.getImageSquarePixels(getContentResolver(), item.uri) > STATUS_MEDIA_PIXEL_SIZE_LIMIT)) {
```

Comment:
I don't quite understand why we need to query media size again tbh

Reply:
There are 2 limits: filesize (8MB) and pixel count (16 mega pixels) and thats two totally different things. Files with less than 8MB but over 16 megapixels caused the bug.